packages/driver/src/cypress.ts

```diff
@@ -194,6 +195,7 @@ class $Cypress {
     this.browser = config.browser
     this.platform = config.platform
     this.testingType = config.testingType
+    this.testFilter = config.testFilter
```

Comment:
Setting this on the Cypress instance like this prob makes sense to run this faster, but a new run finishes on the cloud with updated results, how will this reflect here? I assume the user would need to navigate away from this spec and come back to get the updated test filter?

Also am I understanding the reason for adding this -- to filter down flakey (or is it failed) cloud test results? If that's the case can we name this accordingly? testFilter sounds like a user-provided filter value and doesn't indicate to me this value is driven from cloud data. Would be helpful for maintenance and change impact if this ever needed updated.

Reply:
The idea was to keep the filtering somewhat generic, even though it has only one use case to start.  I would think that the driver would not care why the tests were being filtered, just that it was asked to filter them.  Our thoughts is that this could be expanded in the future to add more ways to filter tests.

Reply:
I can see that...Though it would typically think filter being a regex or list of regex patterns. Which maybe that could be the case later one. The fact it's stored on the global Cypress instance vs state make it feel like this value will be "static" for this Cypress instance. 

This request https://github.com/cypress-io/cypress/issues/4886, would likely use some of this logic if we chose to support them, which would be driven by user actions in the reporter and not necessarily create a new Cypress instance. I also feel like there was another request recently for selecting a single test in the command long to run in `.only()` mode for quick debugging - just couldn't find the issue.

Reply:
I think generating a new Cypress instance would be the cleanest solution as we already have all the wiring for restarting a spec (spec file changed, hitting the refresh button). Might be less performant but restarting has never been too expensive to my knowledge. When we restart, we get another pass at the mocha suite which allows the `testFilter` to get applied again. This testFilter could be modified with each cycle, enabling [cypress-io/cypress/issues/4886](https://github.com/cypress-io/cypress/issues/4886) by listening to an event capturing the fun results and changing the filter based on what failed, then triggering a restart.

I could change the filter to be regex/callback rather than string array which would add more utility to it.